pythonforandroid/recipes/python3/__init__.py

```diff
@@ -71,15 +71,16 @@ class Python3Recipe(TargetPythonRecipe):
         # Python 3.8.1 & 3.9.X
         ('patches/py3.8.1.patch', version_starts_with("3.8")),
         ('patches/py3.8.1.patch', version_starts_with("3.9")),
-        ('patches/py3.8.1.patch', version_starts_with("3.10"))
+        ('patches/py3.8.1.patch', version_starts_with("3.10")),
```

Comment:
Seems pretty strange that we do not have to patch for the ctypes library.

Did you already performed some tests?

Reply:
Are you talking about #2840

Reply:
Please @T-Dynamos do not resolve conversation, unless are really resolved, otherwise I might lose something during the process. (As this one)

I'm not talking about #2840 (which is super good if we fixed it, but should be handled separately).

I was talking about this code snippet into `patches/py3.8.1.patch`, which has been applied on all the recent versions (and seems strange we do not need it on `3.11`):
```
+# This patch overrides the find_library to look in the right places on
+# Android
+if True:
+    from android._ctypes_library_finder import find_library as _find_lib
+    def find_library(name):
+        return _find_lib(name)
```

I'm starting to perform some on-device tests, and I will complete them by the end of the day.

Reply:
So should I apply it?

Reply:
Just finished this test.

Yes, as `ctypes.util.find_library('c')` now returns `None`, instead a path should be returned.

Please be aware that the old patch is not compatible, so a new one need to be produced.

Reply:
```patch
--- Python-3.11.5/Lib/ctypes/util.py	2023-08-24 17:39:18.000000000 +0530
+++ Python-3.11.5.mod/Lib/ctypes/util.py	2023-11-18 21:51:50.167154465 +0530
@@ -4,7 +4,12 @@
 import sys
 
 # find_library(name) returns the pathname of a library, or None.
-if os.name == "nt":
+if True:
+    from android._ctypes_library_finder import find_library as _find_lib
+    def find_library(name):
+        return _find_lib(name)
+
+elif os.name == "nt":
 
     def _get_build_version():
         """Return the version of MSVC that was used to build Python.
```

Reply:
`cpython-311-ctypes-find-library.patch` ? (should only be applied to `3.11` ATM)

Maybe let's also add a comment while using the patch that explains this is needed for `ctypes.util.find_library('c')` to work.